File: icore/src/main/java/com/ddu/icore/bezier/OnePowerPointFTypeEvaluator.java
package com.ddu.icore.bezier;

import android.animation.TypeEvaluator;
import android.graphics.PointF;

/**
 * Created by yzbzz on 2019-08-05.
 */
public class OnePowerPointFTypeEvaluator implements TypeEvaluator<PointF> {

    @Override
    public PointF evaluate(float fraction, PointF startValue, PointF endValue) {
        return BezierUtils.CalculateBezierPointForOnePower(fraction, startValue, endValue);
    }
}

File: icore/src/main/java/com/ddu/icore/bezier/OnePowerPointFTypeEvaluator.kt
package com.ddu.icore.bezier

import android.animation.TypeEvaluator
import android.graphics.PointF

/**
 * Created by yzbzz on 2019-08-05.
 */
class OnePowerPointFTypeEvaluator : TypeEvaluator<PointF> {

    override fun evaluate(fraction: Float, startValue: PointF, endValue: PointF): PointF {
        return BezierUtils.calculateBezierPointForOnePower(fraction, startValue, endValue)
    }
}
